bin/zowe-install-component.sh

```diff
@@ -0,0 +1,339 @@
+#                     - auto: only tag when manifest is in ISO8859-1 encoding.
+#                     This option is only applicable for z/OS.
+# -k|--core           optional boolean. Whether this component is bundled
+#                     into Zowe package.
+# -l|--logs-dir       optional. path to logs directory.
+# -f|--log-file       optional. write log to the file specified.
+#######################################################################
+
+#######################################################################
+# Constants
+DEFAULT_TARGET_DIR=/global/zowe/extensions
+
+#######################################################################
+# Prepare shell environment
+if [ -z "${ZOWE_ROOT_DIR}" ]; then
+  export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
+fi
+
+. ${ZOWE_ROOT_DIR}/bin/internal/zowe-set-env.sh
+
+. ${ZOWE_ROOT_DIR}/bin/utils/utils.sh
+
+# node is required for read_component_manifest
+if [ -n "${NODE_HOME}" ]; then
+  ensure_node_is_on_path
+fi
+
+# is on z/OS?
+RUN_ON_ZOS=$(test `uname` = "OS/390" && echo "true")
+
+#######################################################################
+# Functions
+error_handler(){
+    print_error_message "$1"
+    exit 1
+}
+
+prepare_log_file() {
+    if [ -z "${LOG_FILE}" ]; then
+        set_install_log_directory "${LOG_DIRECTORY}"
+        validate_log_file_not_in_root_dir "${LOG_DIRECTORY}" "${ZOWE_ROOT_DIR}"
+        set_install_log_file "zowe-install-component"
+    else
+        set_install_log_file_from_full_path "${LOG_FILE}"
+        validate_log_file_not_in_root_dir "${LOG_FILE}" "${ZOWE_ROOT_DIR}"
+    fi
+}
+
+extract_to_target_dir(){
+    cd "${TARGET_DIR}" && rm -fr temp-ext-dir
+
+    if [ -d "${COMPONENT_FILE}" ]; then
+        log_message "- component is a directory, create symbolic link to target directory."
+        ln -s "${COMPONENT_FILE}" temp-ext-dir
+    else
+        # create temporary directory to lay down extension files in
+        mkdir -p temp-ext-dir && cd temp-ext-dir
+
+        log_message "- extract file $(basename ${COMPONENT_FILE}) to temporary directory."
+
+        if [[ "$COMPONENT_FILE" = *.pax ]]; then
+            pax -ppx -rf "$COMPONENT_FILE"
+        elif [[ "$COMPONENT_FILE" = *.zip ]]; then
+            jar xf "$COMPONENT_FILE"
+        elif [[ "$COMPONENT_FILE" = *.tar ]]; then
+            pax -z tar -xf "$COMPONENT_FILE"
+        fi
+    fi
+
+    # automatically tag files
+    if [ "${RUN_ON_ZOS}" = "true" ]; then
+        manifest_encoding=
+        if [ "${AUTO_ENCODING}" = "auto" ]; then
+            manifest_encoding=$(detect_component_manifest_encoding "${TARGET_DIR}/temp-ext-dir")
+        fi
+        if [ "${AUTO_ENCODING}" = "yes" -o "${manifest_encoding}" = "ISO8859-1" ]; then
+            # automatically tag files
+            log_message "- automatically tag files"
+            $ZOWE_ROOT_DIR/scripts/utils/tag-files.sh "${TARGET_DIR}/temp-ext-dir" 1>&2 >> $LOG_FILE
```

Comment:
i hope our documentation will explain that `tag-files.sh` is opinionated about which extensions should be in which encoding. There is more than one right answer, but `tag-files.sh` chooses one.

Reply:
Added to docs site PR1526.